Tests/CombineXRexTests/SwiftUI/ObservableViewModelTests.swift

```diff
@@ -0,0 +1,333 @@
+        // Can't test objectWillChange Publisher because it happens before the mutation
+        let subscription = viewModel.statePublisher.sink { [unowned self] state in
+            switch count {
+            case 0:
+                XCTAssertEqual("name: Initial State", state)
+            case 1:
+                XCTAssertEqual("name: Initial State_a1", state)
+            case 2:
+                XCTAssertEqual("name: Initial State_a1_ma:a1", state)
+            case 3:
+                XCTAssertEqual("name: Initial State_a1_ma:a1_maar:a1", state)
+                shouldBeNotified.fulfill()
+            default:
+                XCTFail("Unexpected notification: \(self.viewModel.state)")
+            }
+            count += 1
+        }
+        viewModel.dispatch(.event1(Event1()), from: .here())
+
+        wait(for: [shouldBeNotified], timeout: 1)
+        XCTAssertNotNil(subscription)
+    }
+
+    func testWillChangeNotifyOnChangeIntegrationTest() {
+        let shouldBeNotifiedByWillChangePublisher = expectation(description: "should be notified by will change publisher")
+        var count = 0
+
+        let subscription = viewModel.objectWillChange.sink { [unowned self] _ in
+            switch count {
+            // expected one notification less (only changes, not initial state) and always with the previous value
+            // not yet the one being set.
+            case 0:
+                XCTAssertEqual("name: Initial State", self.viewModel.state)
+            case 1:
+                XCTAssertEqual("name: Initial State_a1", self.viewModel.state)
+            case 2:
+                XCTAssertEqual("name: Initial State_a1_ma:a1", self.viewModel.state)
+                shouldBeNotifiedByWillChangePublisher.fulfill()
+            default:
+                XCTFail("Unexpected notification: \(self.viewModel.state)")
+            }
+            count += 1
+        }
+        viewModel.dispatch(.event1(Event1()), from: .here())
+
+        wait(for: [shouldBeNotifiedByWillChangePublisher], timeout: 1)
+        XCTAssertNotNil(subscription)
+    }
+
+    func testObservableViewModelShouldNotLeak() {
+        weak var obVMWeakRef: ObservableViewModel<String, String>?
+        weak var storeWeakRef: ReduxStoreBase<String, String>?
+
+        autoreleasepool {
+            let store = ReduxStoreBase(
+                subject: .combineX(initialValue: ""),
+                reducer: Reducer<String, String>.identity,
+                middleware: IdentityMiddleware<String, String, String>(),
+                emitsValue: .whenDifferent
+            )
+            storeWeakRef = store
+
+            let obVMStrongRef = store.asObservableViewModel(initialState: "")
+            obVMWeakRef = obVMStrongRef
+            XCTAssertNotNil(obVMWeakRef)
+        }
+
+        XCTAssertNil(storeWeakRef)
+        XCTAssertNil(obVMWeakRef, "middleware should be freed")
+    }
+
+//    func testWillChangePublisherCanHaveMultipleSubscriptions() {
```

Comment:
Remove ?

Reply:
This is a good reminder that I should work on tests and not new features! Guilty!
I'll focus on tests, including to CombineX support, otherwise the coverage will drop even more. :(